### PR TITLE
MAGENTO2-156

### PR DESCRIPTION
**Pull request by Skistims**
fix inconsistent usage of createAddressData

### CODE REVIEW

File: src/Helper/Address.php [hunk 1]
```diff
@@ -58,6 +58,11 @@ public function exists($customerId, array $addressToCheck)
             ->getCollection()
             ->addFieldToFilter('parent_id', $customerId);
 
+        if (empty($addressToCheck['company'])) {
```
Comment (augsteyer) [inline src/Helper/Address.php]: move this up with the other `addressCheck` manipulation, nicer to have these things in one place if this function grows more.

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/) and this p
  - Incompatibility with Magento 2 SOAP API
  - Option validation for child products
  - Corrected order addresses to pass validation
+ - saving of addresses in order import
```
Comment (augsteyer) [inline CHANGELOG.md]: capitalized

**Review by augsteyer — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (Skistims) [reply CHANGELOG.md]: done

File: src/Helper/Address.php (hunk 1 above)
Comment (Skistims) [reply src/Helper/Address.php]: makes sense. done

File: src/Helper/Address.php [hunk 3]
```diff
@@ -54,10 +54,15 @@ public function __construct(AddressFactory $addressFactory)
...
         unset($addressToCheck['email']);
+        if (empty($addressToCheck['company'])) {
+            // company would be cast to an empty string, which does not match NULL in the database
+            unset($addressToCheck['company']);
+        }
         $addressCollection = $this->addressFactory->create()
             ->getCollection()
             ->addFieldToFilter('parent_id', $customerId);
 
+
```
Comment (augsteyer) [inline src/Helper/Address.php]: khem, 2 lines now :)

**Review by augsteyer — CHANGES_REQUESTED** [review]

**Review by augsteyer — APPROVED** [review]

**Comment by augsteyer** [comment]
@Skistims for the future, I wouldn't recommend merging before the travis is done, even if I approve :P ya never know what might fail!

**Comment by Skistims** [comment]
I would agree, if the travis wouldn't fail anyway.
hopefully we will ge this fixed after the release

**Comment by augsteyer** [comment]
it just fails on the PR branch, it does not fail on individual commits you are making, e.g. https://travis-ci.org/shopgate/cart-integration-magento2-base/builds

**Pull request merged**